Library/Homebrew/rubocops/formula_desc_cop.rb

```diff
@@ -0,0 +1,78 @@
+require_relative "./extend/formula_cop"
+require_relative "../extend/string"
+
+module RuboCop
+  module Cop
+    module Homebrew
+      # This cop audits `desc` in Formulae
+      #
+      # - Checks for existence of `desc`
+      # - Checks if size of `desc` > 80
+      # - Checks if `desc` begins with an article
+      # - Checks for correct usage of `command-line` in `desc`
+      # - Checks if `desc` contains the formula name
+      class FormulaDesc < FormulaCop
+        def audit_formula(node, class_node, _parent_class_node, body)
```

Comment:
Ideally this method would contain code like the following:
```ruby
def audit_formula
  desc_call = find_method_call(:desc)
  return unless desc_call
  desc = parameters(desc_call).first

  max_length = 80
  if desc.length > max_length
    problem <<-EOS.undent
      Description is too long. "name: desc" should be less than #{max_length} characters.
      Length is calculated as #{formula_name} + desc. (currently #{"#{formula_name}: #{desc}".length})
    EOS

  if desc =~ /(command ?line)/i
    problem "Description should use \"command-line\" instead of \"#{$1}\""
  end

 ... 

end
```

The `find_method_call` and `parameters?` methods should be defined in `formula_cop.rb` and the various instance variables set by `on_class` or something it calls and by e.g. `find_method_call` and `parameters` so `problem` can access e.g. the `source_range`.

Obviously this can (and will) vary but just giving you a heads up for the level of complexity I think would be ideal for these cops to hide the implementation details.